### ISSUE TITLE
[Bug]: iOS Bundle identifier, App Group mismatch

### ISSUE THREAD

**Issue by yusufkildan** [OneSignal/OneSignal-Unity-SDK#545]
### What happened?

If the previous post-process scripts change the bundle identifier, OneSignal's BuildPostProcessor script uses the wrong bundle identifier and it is causing CI build errors.

### Steps to reproduce?

```Markdown
1. Create post-process script with 0 callback order.
2. Change PlayerSettings.applicationIdentifier with script.
3. Run OneSignal's BuildPostProcessor script.

_appGroupName field of com.onesignal.unity.ios/Editor/BuildPostProcessor.cs uses old bundle identifier.
```

### What did you expect to happen?

Solution suggestion:
`private readonly string _appGroupName = $"group.{PlayerSettings.applicationIdentifier}.onesignal";`
can be updated as property instead of field:
`private string _appGroupName => $"group.{PlayerSettings.applicationIdentifier}.onesignal";`

### Unity version

2020.3.35.f1

### OneSignal Unity SDK version

3.0.4

### Platform

iOS

### Relevant log output

```Shell
❌  error: Provisioning profile "Bitrise iOS development - (com.x.x.dev.OneSignalNotificationServiceExtension)" doesn't support the group.com.x.x.onesignal App Group. (in target 'OneSignalNotificationServiceExtension' from project 'Unity-iPhone')
❌  error: Provisioning profile "Bitrise iOS development - (com.x.x.dev.OneSignalNotificationServiceExtension)" doesn't match the entitlements file's value for the com.apple.security.application-groups entitlement. (in target 'OneSignalNotificationServiceExtension' from project 'Unity-iPhone')
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shepherd-l** [comment]
Thanks for the suggestion! We plan to add this change in a future update.

**Comment by shepherd-l** [comment]
Added this fix in the [3.0.6](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/3.0.6) release thanks to @EtienneUku
